heracles/result.py

```diff
@@ -291,7 +296,21 @@ def truncated(result, ell_max):
 
     return Result(
         out,
+        spin=result.spin,
         ell=truncated_ell,
         axis=axes,
         weight=truncated_weight,
     )
+
+
+def _update_result_array(result, arr):
+    """Return a copy of result with array replaced by arr."""
+    return Result(
+        arr,
+        spin=result.spin,
+        ell=result.ell,
+        axis=result.axis,
+        lower=result.lower,
+        upper=result.upper,
+        weight=result.weight,
+    )
```

Comment:
Instead of this new function, we should use the default `replace()` mechanism for that (which is now a standard in Python 3.13+).

```py
try:
    from copy import replace
except AttributeError:
    # Python < 3.13
    from dataclasses import replace

replace(result, array=arr)
```